scripts/Diffraction/isis_powder/routines/common.py

```diff
@@ -156,7 +157,10 @@ def get_first_run_number(run_number_string):
     :return: The first run for the user input of runs
     """
     run_numbers = generate_run_numbers(run_number_string=run_number_string)
-    return run_numbers[0]
+    if isinstance(run_numbers, list):
```

Comment:
also in `get_monitor_ws` [here](https://github.com/mantidproject/mantid/pull/19163/files#diff-3e56862968e7fc4f56f839cc920d48baL174), I think you could use get_first_run_number